src/common/transformations/src/ov_ops/moe_expert.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ov_ops/moe_expert.hpp"
+
+#include "itt.hpp"
+
+namespace ov {
+namespace op {
+namespace internal {
+
+MOEExpert::MOEExpert(const OutputVector& args, const Config& cfg, const std::vector<ConstsPerExpert>& consts)
+    : Op(args),
+      m_config(cfg),
+      m_consts(consts) {
+    constructor_validate_and_infer_types();
+}
+
+const MOEExpert::Config& MOEExpert::get_config() const {
+    return m_config;
+}
+
+void MOEExpert::set_config(const Config& config) {
+    m_config = config;
+}
+
+std::shared_ptr<ov::Node> MOEExpert::clone_with_new_inputs(const ov::OutputVector& new_args) const {
+    INTERNAL_OP_SCOPE(internal_MOEExpert_clone_with_new_inputs);
+    check_new_args_count(this, new_args);
+
+    return std::make_shared<MOEExpert>(new_args, m_config, m_consts);
+}
+
+void MOEExpert::validate_and_infer_types() {
+    INTERNAL_OP_SCOPE(internal_MOEExpert_validate_and_infer_types);
+    OPENVINO_ASSERT(get_input_size() == 2 || get_input_size() == 4,
+                    "MOEExpert must have 2/4 inputs whereas it has ",
+                    get_input_size());
+    OPENVINO_ASSERT(get_output_size() == 1, "MOEExpert must have 1 output whereas it has ", get_output_size());
```

Comment:
Here output size may be 0 the outputs are added next line

Reply:
Fixed.